hugegraph-core/src/main/java/org/apache/hugegraph/backend/tx/EphemeralJobQueue.java

```diff
@@ -0,0 +1,236 @@
+                this.state.compareAndSet(State.EXECUTE, State.INIT);
+            }
+        }
+    }
+
+    public boolean isEmpty() {
+        return this.pendingQueue.isEmpty();
+    }
+
+    public static class BatchEphemeralJob extends EphemeralJob<Object> {
+
+        private static final int PAGE_SIZE = 100;
+        private static final String BATCH_EPHEMERAL_JOB = "batch-ephemeral-job";
+        private static final int MAX_CONSUME_COUNT = EphemeralJobQueue.CAPACITY / 2;
+
+        private WeakReference<EphemeralJobQueue> queueWeakReference;
+
+        public BatchEphemeralJob(EphemeralJobQueue queue) {
+            this.queueWeakReference = new WeakReference<>(queue);
+        }
+
+        @Override
+        public String type() {
+            return BATCH_EPHEMERAL_JOB;
+        }
+
+        @Override
+        public Object execute() throws Exception {
+            boolean stop = false;
+            Object ret = null;
+            int consumeCount = 0;
+            InterruptedException interruptedException = null;
+            EphemeralJobQueue queue;
+            List<EphemeralJob<?>> batchJobs = new ArrayList<>();
+            while (!stop) {
+                if (interruptedException == null && Thread.currentThread().isInterrupted()) {
+                    interruptedException = new InterruptedException();
+                }
+
+                queue = this.queueWeakReference.get();
+                if (queue == null) {
+                    stop = true;
+                    continue;
+                }
+
+                if (queue.isEmpty() || consumeCount > MAX_CONSUME_COUNT ||
+                    interruptedException != null) {
+                    queue.consumeComplete();
+                    stop = true;
+                    if (!queue.isEmpty()) {
+                        queue.reScheduleIfNeeded();
+                    }
+                    continue;
+                }
+
+                try {
+                    while (!queue.isEmpty() && batchJobs.size() < PAGE_SIZE) {
+                        EphemeralJob<?> job = queue.queue().poll();
+                        if (job == null) {
+                            continue;
+                        }
+                        batchJobs.add(job);
+                        consumeCount++;
+                    }
+
+                    if (batchJobs.isEmpty()) {
+                        continue;
+                    }
+
+                    ret = this.executeBatchJob(batchJobs, ret);
+
+                } catch (InterruptedException e) {
+                    interruptedException = e;
+                } finally {
+                    batchJobs.clear();
+                }
+            }
+
+            if (interruptedException != null) {
+                Thread.currentThread().interrupt();
+                throw interruptedException;
+            }
+
+            return ret;
+        }
+
+        private Object executeBatchJob(List<EphemeralJob<?>> jobs, Object prev) throws Exception {
```

Comment:
can keep List\<Job\<?>>?

Reply:
the Queue main to batch EphemeralJob, so the name is EphemeralJob perfer, and need to set params value when init the job(SysTaskCallable define the params method)